src/libexpr/tests/value/context.cc

```diff
@@ -95,7 +95,7 @@ Gen<NixStringContextElem::Built> Arbitrary<NixStringContextElem::Built>::arbitra
 
 Gen<NixStringContextElem> Arbitrary<NixStringContextElem>::arbitrary()
 {
-    switch (*gen::inRange<uint8_t>(0, 2)) {
+    switch (*gen::inRange<uint8_t>(0, std::variant_size_v<NixStringContextElem::Raw>)) {
```

Comment:
To make this fail safe, the switches should end in

```c++
default:
    assert(false);
```

Alternatively this could be abstracted to a remotely `std::visit`-like function that forces the shape statically, but that'd take some proper wizardry.